src/modules/google-organization-kp/includes/Publisher_Service.php

```diff
@@ -0,0 +1,283 @@
+			$type_uri = sprintf( 'http://schema.org/%s', $params['type'] );
+
+			$this->entity_service->set(
+				$publisher_id,
+				$type_uri,
+				true
+			);
+		}
+
+		// Update Alternate Name.
+		if ( isset( $params['alt_name'] ) ) {
+			$this->configuration_service->set_alternate_name( sanitize_text_field( $params['alt_name'] ) );
+		}
+
+		// Set the entity logo.
+		if ( isset( $params['image'] ) ) {
+			$image_file  = $params['image'];
+			$upload_dir  = wp_upload_dir();
+			$target_dir  = $upload_dir['path'];
+			$target_file = $target_dir . '/' . basename( $image_file['name'] );
+
+			if ( move_uploaded_file( $image_file['tmp_name'], $target_file ) ) {
+				$attachment = array(
+					'post_title'     => basename( $target_file ),
+					'post_mime_type' => wp_check_filetype( $target_file )['type'],
+					'post_status'    => 'inherit',
+					'post_parent'    => $publisher_id,
+				);
+
+				$attachment_id = wp_insert_attachment( $attachment, $target_file, $publisher_id );
+				set_post_thumbnail( $publisher_id, $attachment_id );
+			} else {
+				throw new Exception( 'Unable to update the image' );
+			}
+		}
+
+		// Update fields.
+
+		$schema_service = $this->schema_service;
+
+		if ( isset( $params['same_as'] ) ) {
+			update_post_meta( $publisher_id, $schema_service::FIELD_SAME_AS, $params['same_as'] );
+		}
+
+		if ( isset( $params['address'] ) ) {
+			update_post_meta( $publisher_id, $schema_service::FIELD_ADDRESS, $params['address'] );
+		}
+
+		if ( isset( $params['locality'] ) ) {
+			update_post_meta( $publisher_id, $schema_service::FIELD_ADDRESS_LOCALITY, $params['locality'] );
+		}
+
+		if ( isset( $params['region'] ) ) {
+			update_post_meta( $publisher_id, $schema_service::FIELD_ADDRESS_REGION, $params['region'] );
+		}
+
+		if ( isset( $params['country'] ) ) {
+			update_post_meta( $publisher_id, $schema_service::FIELD_ADDRESS_COUNTRY, $params['country'] );
+		}
+
+		if ( isset( $params['postal_code'] ) ) {
+			update_post_meta( $publisher_id, $schema_service::FIELD_ADDRESS_POSTAL_CODE, $params['postal_code'] );
+		}
+
+		if ( isset( $params['telephone'] ) ) {
+			update_post_meta( $publisher_id, $schema_service::FIELD_TELEPHONE, $params['telephone'] );
+		}
+
+		if ( isset( $params['email'] ) ) {
+			update_post_meta( $publisher_id, $schema_service::FIELD_EMAIL, $params['email'] );
+		}
+
+		if ( isset( $params['no_employees'] ) ) {
+			update_post_meta( $publisher_id, $schema_service::FIELD_NO_OF_EMPLOYEES, $params['no_employees'] );
+		}
+
+		if ( isset( $params['founding_date'] ) ) {
+			update_post_meta( $publisher_id, $schema_service::FIELD_FOUNDING_DATE, $params['founding_date'] );
+		}
+
+		if ( isset( $params['iso_6523'] ) ) {
+			update_post_meta( $publisher_id, $schema_service::FIELD_ISO_6523_CODE, $params['iso_6523'] );
+		}
+
+		if ( isset( $params['naics'] ) ) {
+			update_post_meta( $publisher_id, $schema_service::FIELD_ISO_6523_CODE, $params['naics'] );
+		}
+
+		if ( isset( $params['global_loc_no'] ) ) {
```

Comment:
What If I remove the value, then you won't unset it? What I set an empty value? you won't delete the meta?